pkg/handler/project/project.go

```diff
@@ -1030,3 +1034,140 @@ func (h *handler) UpdateGeneralInfo(c *gin.Context) {
+	projectID := c.Param("id")
+	if projectID == "" || !model.IsUUIDFromString(projectID) {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrInvalidProjectID, nil, ""))
+		return
+	}
+
+	var body UpdateContactInfoInput
+	if err := c.ShouldBindJSON(&body); err != nil {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler": "project",
+		"method":  "UpdateContactInfo",
+		"id":      projectID,
+		"request": body,
+	})
+
+	// Check project exists
+	exists, err := h.store.Project.Exists(h.repo.DB(), projectID)
+	if err != nil {
+		l.Error(err, "error when check existence of project")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	if !exists {
+		l.Error(err, "project not found")
+		c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrProjectNotFound, body, ""))
+		return
+	}
+
+	// Check account manager exists
+	exists, err = h.store.Employee.Exists(h.repo.DB(), body.AccountManagerID.String())
+
+	if err != nil {
+		l.Error(err, "error when finding account manager")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	if !exists {
+		l.Error(ErrAccountManagerNotFound, "account manager not found")
+		c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrAccountManagerNotFound, body, ""))
+		return
+	}
+
+	// Check delivery manager exists
+	exists, err = h.store.Employee.Exists(h.repo.DB(), body.DeliveryManagerID.String())
+
+	if err != nil {
+		l.Error(err, "error when finding delivery manager")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	if !exists {
+		l.Error(ErrDeliveryManagerNotFound, "delivery manager not found")
+		c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrDeliveryManagerNotFound, body, ""))
+		return
+	}
+
+	// Begin transaction
+	tx, done := h.repo.NewTransaction()
+
+	// Delete account manager and delivery manager if exists
+	if err := h.store.ProjectHead.DeleteByProjectIDAndPosition(tx.DB(), projectID, model.HeadPositionAccountManager.String()); err != nil {
+		l.Error(err, "failed to delete account manager of project in db")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), body, ""))
+		return
+	}
+
+	if err := h.store.ProjectHead.DeleteByProjectIDAndPosition(tx.DB(), projectID, model.HeadPositionDeliveryManager.String()); err != nil {
+		l.Error(err, "failed to delete delivery manager of project in db")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), body, ""))
+		return
+	}
+
+	// Create new one
+	newAccountManager := model.ProjectHead{
+		ProjectID:  model.MustGetUUIDFromString(projectID),
+		EmployeeID: body.AccountManagerID,
+		JoinedDate: time.Now(),
```

Comment:
time now should be defined as a variable. timeNow := time.Now() then use this value for all function

Reply:
updated